Automattic/wp-calypso
# AI Logo Generator: Add dynamic logo cost

Related to #85557

## Proposed Changes

* Uses `feature.costs.<feature-name>.action` from the Jetpack AI feature data to determine the cost of a logo
* Shows the upgrade nudge when there are no enough requests left and a different modal for first request

## Testing Instructions



The cost defaults to 10 to ensure it works before the data is provided by the backend
* With a tiered site, use `add_filter( 'jetpack_ai_current_period_requests_count', function() { return X; } );` to your `0-sandbox.php` file and sandbox the public API.
* Modify the X value to less than 11
* Without any history, you should see a modal asking for an upgrade (as that case requires 11 requests)
* Modify the X value to less than 10
* With a previous history, check that the upgrade nudge is shown when the modal is opened

## Pre-merge Checklist
